ServerCore/Helpers/MailHelper.cs

```diff
@@ -182,14 +182,24 @@ private void SendInternal2(IEnumerable<string> recipients, bool bcc, string subj
             AddRecipients(msg.Bcc, extraBcc);
         }
 
-        SmtpClient client = new SmtpClient("in.mailjet.com", 587);
-        // Unlike port 25 which is for general SMTP, port 587 is for
-        // MSA (Message Submission Agent) use and is authenticated.
-        client.DeliveryMethod = SmtpDeliveryMethod.Network;
-        client.EnableSsl = true;
-        client.UseDefaultCredentials = false;
-        client.Credentials = new NetworkCredential(PublicSecret, PrivateSecret);
-        client.SendMailAsync(msg);
+        if (msg.To.Count > 0 || msg.CC.Count > 0 || msg.Bcc.Count > 0)
+        {
+            SmtpClient client = new SmtpClient("in.mailjet.com", 587);
+            // Unlike port 25 which is for general SMTP, port 587 is for
+            // MSA (Message Submission Agent) use and is authenticated.
+            client.DeliveryMethod = SmtpDeliveryMethod.Network;
+            client.EnableSsl = true;
+            client.UseDefaultCredentials = false;
+            client.Credentials = new NetworkCredential(PublicSecret, PrivateSecret);
+            client.SendMailAsync(msg);
+        }
+        else
+        {
+            if (!IsDev)
```

Comment:
Why are you writing a debug log message only in production? I would think you would do the reverse.

Reply:
I copied and pasted this code from #426 but I can reverse it.